### PR TITLE
Refactoring of LanguageModelProvider and LanguageModelProviderRegistry

### PR DESCRIPTION
**Pull request by planger**
#### What it does

https://github.com/eclipsesource/osweek-2024/issues/47

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-core/src/node/language-model-frontend-delegate.ts [hunk 1]
```diff
@@ -17,55 +17,60 @@
...
+    setClient(client: LanguageModelDelegateClient): void {
         this.frontendDelegateClient = client;
     }
 
     async request(
         modelId: string,
         request: LanguageModelRequest
     ): Promise<LanguageModelResponseDelegate> {
-        const provider = await this.registry.getLanguageModelProvider(modelId);
+        const provider = await this.registry.getLanguageModel(modelId);
```
Comment (eneufeld) [inline packages/ai-core/src/node/language-model-frontend-delegate.ts:73]: the provider should be renamed too

**Review by eneufeld — COMMENTED** [review]
schaut gut aus, der language-model-provider zu language-model rename ist jetzt lost weil es da noch changes gab

File: packages/ai-core/src/node/language-model-frontend-delegate.ts (hunk 1 above)
Comment (planger) [reply packages/ai-core/src/node/language-model-frontend-delegate.ts]: Done

**Review by sdirix — APPROVED** [review]
LGTM

**Pull request merged**